File: covid-19-mobile/app/src/main/java/pl/mil/wp/help/API/models/UserHealthStatus.java
package pl.mil.wp.help.API.models;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

/**
 * Gets or Sets UserHealthStatus
 */
@JsonAdapter(UserHealthStatus.Adapter.class)
public enum UserHealthStatus {
  
  HEALTHY("healthy"),
  
  RECOVERED("recovered"),
  
  QUARANTINE("quarantine"),
  
  DEATH("death");

  private String value;

  UserHealthStatus(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static UserHealthStatus fromValue(String text) {
    for (UserHealthStatus b : UserHealthStatus.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    return null;
  }

  public static class Adapter extends TypeAdapter<UserHealthStatus> {
    @Override
    public void write(final JsonWriter jsonWriter, final UserHealthStatus enumeration) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public UserHealthStatus read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return UserHealthStatus.fromValue(String.valueOf(value));
    }
  }
}

File: covid-19-backend/CovidHelp/angular/src/app/generated-api/models/user-health-status.ts
/* tslint:disable */
type UserHealthStatus =
  'healthy' |
  'recovered' |
  'quarantine' |
  'death';
module UserHealthStatus {
  export const HEALTHY: UserHealthStatus = 'healthy';
  export const RECOVERED: UserHealthStatus = 'recovered';
  export const QUARANTINE: UserHealthStatus = 'quarantine';
  export const DEATH: UserHealthStatus = 'death';
  export function values(): UserHealthStatus[] {
    return [
      HEALTHY,
      RECOVERED,
      QUARANTINE,
      DEATH
    ];
  }
}

export { UserHealthStatus }